src/Models/DataObjectTaxonomyTerm.php

```diff
@@ -32,4 +32,23 @@ class DataObjectTaxonomyTerm extends DataObject
     private static $extensions = [
         Versioned::class,
     ];
+
+
+    /**
+     * Though this object is used to link {@see DataObject} and {@see TaxonomyTerm} in a many-many relation, the
+     * object itself is manipulated as a data object and is {@see Versioned} by default, it could be published by its
+     * owner object through the static $owns {@see DataObjectTaxonomiesDataExtension::$owns} defined. On an operation
+     * of deleting this object trigged by 'unlink' on its owner's interface, this object is removed only from its
+     * Staging stage, whereas leaves its published record as an orphaned record. The function is to ensure the object's
+     * published record is also removed to maintains data integrity.
+     */
+    public function onAfterDelete()
+    {
+        if ($this->hasExtension(Versioned::class)) {
```

Comment:
Literally above this method we add the extension, so we don't need to check if it's applied.

Reply:
yes, but the module user might remove it for their specific project. As we are manipulating records which are instances rather than class-level settings, so better check it before calling the function `isPublished()`. 
It will make a project can't  'unlink' this relation if the remove the Versioned from the bridging data object.

Reply:
Yes, they can also override this method or not use the extension at all or do any other random stuff. If someone tampers with 3rd party module's code, they can expect issues. The check is unnecessary in my opinion.